cmd/fakemachine/main.go

```diff
@@ -79,6 +84,7 @@ func main() {
 	}
 
 	m := fakemachine.NewMachine()
+	m.SetBackend(options.Backend)
```

Comment:
Adding API to *change* the backend after creating the machine seems both unnecessary as well as risky.. Why not just have a new `NewMachineWithBackend` function to do it in one shot (and have NewMachine call that with e.g. "auto")